weave/trace_server/actions_worker/actions/contains_words.py

```diff
@@ -0,0 +1,24 @@
+import json
+from typing import Any
+
+from weave.trace_server.interface.base_object_classes.actions import (
+    ContainsWordsActionSpec,
+)
+from weave.trace_server.trace_server_interface import (
+    CallSchema,
+    TraceServerInterface,
+)
+
+
+def do_contains_words_action(
+    project_id: str,
+    config: ContainsWordsActionSpec,
+    call: CallSchema,
+    trace_server: TraceServerInterface,
+) -> Any:
```

Comment:
A few questions here:
1. Will actions always be functions with this shape?
2. Will users be able to define their own actions?
3. Would it ever be preferable to write actions as classes?

Reply:
1. yes, we might add to this in the future - it is purely internal implementation detail, so allowed to change
2. No. Instead they would define their own op and define an action that hits that op.
3. Maybe. again, this is all backend implementation for now. I was just keeping it simple for now until we have more examples to pattern match

Reply:
RE:2 -- How do you define "an action that hits (an) op"?
RE:3 -- Ok.  I was initially concerned because you have a strict type def below, but we have total control over this part of the code so we can just change it later

Reply:
#2 -> this is not supported yet, but is in the plan. It will be a new action type ("op_execute") maybe with it's own parameter definition

#3 -> 👍

Reply:
oh I see an action that literally calls the op -- makes sense